src/hotspot/share/services/memReporter.hpp

```diff
@@ -71,27 +71,35 @@ class MemReporterBase : public StackObj {
   }
 
   // Convert diff amount in bytes to current reporting scale
+  // We use int64_t instead of ssize_t because on 32-bit it allows us to express deltas larger than 2 gb.
+  // On 64-bit we never expect memory sizes larger than INT64_MAX.
   int64_t diff_in_current_scale(size_t s1, size_t s2) const {
-    // _scale should not be 0, otherwise division by zero at return.
     assert(_scale != 0, "wrong scale");
 
+    LP64_ONLY(assert(s1 < INT64_MAX, "exceeded possible memory limits");)
+    LP64_ONLY(assert(s2 < INT64_MAX, "exceeded possible memory limits");)
```

Comment:
Nit: LP64_ONLY isn't needed.

Reply:
Would a 32-bit compiler not warn about a comparison that is always true?

Reply:
I don't think so. We don't consider such warnings helpful, because they just
lead to this kind of conditionalization (or dropping such checks altogether).
So far as I know, only clang among widely supported compilers warns about it,
and we've specifically disabled that warning globally because we consider it
non-helpful.

Reply:
Okay, thanks for the clarification. I could have sworn I saw this warning on local x86 builds, but it may have been something like >=0 comparisons for unsigned.